### PR TITLE
feat: switch to csi-proxy GA v1

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: switch to csi-proxy GA v1

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: switch to csi-proxy GA v1
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/container-storage-interface/spec v1.3.0
 	github.com/golang/protobuf v1.4.3
 	github.com/kubernetes-csi/csi-lib-utils v0.7.0
-	github.com/kubernetes-csi/csi-proxy/client v0.2.2
+	github.com/kubernetes-csi/csi-proxy/client v1.0.0-rc.1
```
Comment (mauriciopoppe) [inline go.mod:13]: hey @andyzhangx, FYI we've bumped the rc to v1.0.0, there are no changes in the API and it's just a change in go.mod

Comment (andyzhangx) [reply go.mod:13]: thanks, I will make changes later.

**Pull request merged**